android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -99,39 +104,46 @@ public void run() {
 
     private void loadBundle() {
         mCodePush.clearDebugCacheIfNeeded();
-        final Activity currentActivity = getCurrentActivity();
 
-        if (currentActivity == null) {
-            // The currentActivity can be null if it is backgrounded / destroyed, so we simply
-            // no-op to prevent any null pointer exceptions.
-            return;
-        }
+        // #1) Get the ReactInstanceManager instance, which is what includes the
+        //     logic to reload the current React context.
+        ReactInstanceManager instanceManager = CodePush.getReactInstanceManager();
 
         try {
-            ReactInstanceManager instanceManager;
-            // #1) Get the ReactInstanceManager instance, which is what includes the
-            //     logic to reload the current React context.
-            try {
-                // In RN >=0.29, the "mReactInstanceManager" field yields a null value, so we try
-                // to get the instance manager via the ReactNativeHost, which only exists in 0.29.
-                Method getApplicationMethod = ReactActivity.class.getMethod("getApplication");
-                Object reactApplication = getApplicationMethod.invoke(currentActivity);
-                Class<?> reactApplicationClass = tryGetClass(REACT_APPLICATION_CLASS_NAME);
-                Method getReactNativeHostMethod = reactApplicationClass.getMethod("getReactNativeHost");
-                Object reactNativeHost = getReactNativeHostMethod.invoke(reactApplication);
-                Class<?> reactNativeHostClass = tryGetClass(REACT_NATIVE_HOST_CLASS_NAME);
-                Method getReactInstanceManagerMethod = reactNativeHostClass.getMethod("getReactInstanceManager");
-                instanceManager = (ReactInstanceManager)getReactInstanceManagerMethod.invoke(reactNativeHost);
-            } catch (Exception e) {
-                // The React Native version might be older than 0.29, or the activity does not
-                // extend ReactActivity, so we try to get the instance manager via the
-                // "mReactInstanceManager" field.
-                Class instanceManagerHolderClass = currentActivity instanceof ReactActivity
-                        ? ReactActivity.class
-                        : currentActivity.getClass();
-                Field instanceManagerField = instanceManagerHolderClass.getDeclaredField("mReactInstanceManager");
-                instanceManagerField.setAccessible(true);
-                instanceManager = (ReactInstanceManager)instanceManagerField.get(currentActivity);
+            if (instanceManager == null) {
+                final Activity currentActivity = getCurrentActivity();
+
+                if (!ReactActivity.class.isInstance(currentActivity)) {
```

Comment:
This is a regression of https://github.com/Microsoft/react-native-code-push/pull/470. Could you take a look at that and fix this? Basically we no longer want to immediately fallback to the legacy logic the moment the activity is not a ReactActivity.